_posts/2020-01-07-phpunit-CVE-2017-9841.md

```diff
@@ -0,0 +1,69 @@
+---
+layout: post
+title: "[CVE-2017-9841] PHUnit dependancy in PrestaShop and modules allows remote arbitrary PHP code execution"
+categories: dependancies
+author:
+- Friends-Of-Presta.org
+meta: "CVE,PrestaShop,dependancies"
+severity: 9.8
+---
+
+Modules include the vulnerable dependancy are: 1-Click Upgrade (autoupgrade) from 4.0.0 to 4.10.1, Cart Abandonment Pro (pscartabandonmentpro) from 2.0.1 to 2.0.10, Faceted Search (ps_facetedsearch) from 2.2.1 to 3.4.1, Merchant Expertise (gamification) from 2.1.0 to 2.3.2, PrestaShop Checkout (ps_checkout) from 1.0.8 to 1.2.9.
+
+In addition, if PrestaShop had been initially installed prior to ~1.7.6.0 with composer in dev mode can contain a critical vulnerability if htaccess in vendor directory is not denied.
+
+
+## Summary
+
+* **CVE ID**: [CVE-2017-9841](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2017-9841)
+* **Published at**: 2020-01-20
+* **Advisory source**: [PrestaShop](https://build.prestashop-project.org/news/2020/critical-security-vulnerability-in-prestashop-modules/)
+* **Vendor**: PrestaShop
+* **Product**: PrestaShop
+* **Impacted release**: ~< 1.7.6.0
+* **Product author**: PrestaShop
+* **Weakness**: [CWE-94](https://cwe.mitre.org/data/definitions/94.html)
+* **Severity**: critical (9.8)
+
+## Description
+    
+Util/PHP/eval-stdin.php in PHPUnit before 4.8.28 and 5.x before 5.6.3 allows remote attackers to execute arbitrary PHP code via HTTP POST data beginning with a "<?php " substring, as demonstrated by an attack on a site with an exposed /vendor folder, i.e., external access to the /vendor/phpunit/phpunit/src/Util/PHP/eval-stdin.php URI.
+
+
+## CVSS base metrics
+
+* **Attack vector**: network
+* **Attack complexity**: low
+* **Privilege required**: none
+* **User interaction**: none
+* **Scope**: unchanged
+* **Confidentiality**: high
+* **Integrity**: high
+* **Availability**: high
+
+**Vector string**: CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H
+
+## Possible malicious usage
+
+According to [PrestaShop team](https://build.prestashop-project.org/news/2020/critical-security-vulnerability-in-prestashop-modules/), this vulenrability has been exploited to add a malicious code.
+
+"According to our analysis, most attackers either place new files in the filesystem or modify existing files, like AdminLoginController.php. Here’s a non-exhaustive list of known malicious files that may indicate a compromised shop: XsamXadoo_Bot.php, XsamXadoo_deface.php, 0x666.php, f.php, Xsam_Xadoo.html"
+
+## Patch
+
+Delete the vulnerable file on all your PrestaShop:
+
+```diff
+find . -type d -name "eval-stdin.php" -exec rm -rf {} \;
```

Comment:
in https://build.prestashop-project.org/news/2020/critical-security-vulnerability-in-prestashop-modules/
the command is 
find . -type d -name "phpunit" -exec rm -rf {} \;

Reply:
Yes, it's find . -type f -name "eval-stdin.php" or find . -type f -name "eval-stdin.php"

Reply:
Yep thank you. I update the original fix by PrestaShop because, it doesn't remove the exact vulnerability

Reply:
ok tx
source https://github.com/sebastianbergmann/phpunit/issues/3112#issuecomment-386430232